src/components/QRScannerSubmodal.js

```diff
@@ -0,0 +1,140 @@
+
+      let container = event.target.parentElement
+
+      if (container.classList.contains(styles['scanner-btn'])) {
+        container = container.parentElement
+      }
+
+      if (container.classList.contains(styles['scanner-btn-container'])) {
+        container = container.parentElement
+      }
+
+      container.querySelector(`.${styles['scanner-btn-container']}`).remove()
+      container.querySelector(`.${styles['scanner-container']}`).hidden = false
+      container.querySelector('video').play()
+  }
+
+  render() {
+    const { onScanned } = this.props
+
+    return (
+      <div>
+        <div className={styles['scanner-btn-container']}>
+          <div className={styles['scanner-btn']} onClick={this.openCamera}>
+            <FormattedMessage
+              id="qrcode-scanner.tap-to-scan"
+              defaultMessage="Tap to scan QR Code"
+            />
+          </div>
+        </div>
+
+        <div hidden={true} className={styles['scanner-container']}>
+          <QrReader
+            delay={300}
+            onError={(err) => console.error(err)}
+            onScan={onScanned}
+            className={styles['scanner']}
+          />
+          <div className={styles['scanner-text']}>
+            <FormattedMessage
+              id="add-friend-modal.scan-code-title"
+              defaultMessage="Scan QR Code to add friend"
+            />
+          </div>
+        </div>
+      </div>
+    )
+  }
+}
+
+class IosScanner extends PureComponent {
+  constructor(props) {
+    super();
+    this.openCamera = this.openCamera.bind(this);
+  }
+  openCamera() {
+    let iframe = document.createElement('IFRAME');
+    iframe.setAttribute('src', 'opencamera://');
+
+    window.getQRCode = code => {
+      this.props.onScanned(code);
+      iframe.remove();
+    };
+
+    document.documentElement.appendChild(iframe);
+  }
+  render () {
+    return (
+      <div>
+        <div className={styles['scanner-btn-container']} onClick={this.openCamera}>
+          <div className={styles['scanner-btn']}>
+            <FormattedMessage
+              id="qrcode-scanner.tap-to-scan"
+              defaultMessage="Tap to scan QR Code"
+            />
+          </div>
+        </div>
+      </div>
+    )
+  }
+}
+
+const WebScanner = props => {
+  const { onScanned } = props
+
+  return (
+    <div>
+      <div className={styles['scanner-container']}>
+        <QrReader
+          delay={300}
+          onError={(err) => console.error(err)}
+          onScan={onScanned}
+          className={styles['scanner']}
+        />
+        <div className={styles['scanner-text']}>
+          <FormattedMessage
+            id="add-friend-modal.scan-code-title"
+            defaultMessage="Scan QR Code to add friend"
+          />
+        </div>
+      </div>
+    </div>
+  )
+}
+
+export default props => (
+  <div className={styles['submodal-wrapper']}>
+    {
+      (() => {
+        if (isAndroid()) {
```

Comment:
Is it possible to render more than one scanner when isAndroid and isIOS become non-exclusive?

Reply:
I'll prefer to use `if`, `else if` and `else` for this case.

Reply:
ok, done. :)